src/Entities/Player.cpp

```diff
@@ -2636,3 +2636,101 @@ void cPlayer::FreezeInternal(const Vector3d & a_Location, bool a_ManuallyFrozen)
 	m_FlyingMaxSpeed = FlyingMaxpeed;
 	m_IsFlying = IsFlying;
 }
+
+
+
+
+float cPlayer::GetLiquidHeightPercent(NIBBLETYPE a_Meta)
+{
+	if (a_Meta >= 8)
+	{
+		a_Meta = 0;
+	}
+	return (float)(a_Meta + 1) / 9.0f;
+}
+
+
+
+
+
+bool cPlayer::IsInsideOfBlock(BLOCKTYPE a_Block)
+{
+	BLOCKTYPE Block = m_World->GetBlock(static_cast<int>(GetPosX()), static_cast<int>(m_Stance), static_cast<int>(GetPosZ()));
+	if (Block != a_Block)
+	{
+		return false;
+	}
+	NIBBLETYPE Meta = GetWorld()->GetBlockMeta(static_cast<int>(GetPosX()), static_cast<int>(m_Stance), static_cast<int>(GetPosZ()));
+	float f = GetLiquidHeightPercent(Meta) - 0.11111111f;
+	float f1 = static_cast<float>(m_Stance + 1) - f;
+	bool flag = m_Stance < f1;
+	return flag;
+}
+
+
+
+
+
+float cPlayer::GetDigSpeed(BLOCKTYPE a_Block)
+{
+	float f = GetEquippedItem().GetHandler()->GetStrVsBlock(a_Block);
+	if (f > 1.0f)
+	{
+		unsigned int efficiencyModifier = GetEquippedItem().m_Enchantments.GetLevel(cEnchantments::eEnchantment::enchEfficiency);
+		if (efficiencyModifier > 0)
+		{
+			f += (efficiencyModifier * efficiencyModifier) + 1;
+		}
+	}
+
+	if (HasEntityEffect(cEntityEffect::effHaste))
+	{
+		int intensity = GetEntityEffect(cEntityEffect::effHaste)->GetIntensity() + 1;
+		f *= 1.0f + intensity * 0.2f;
+	}
+
+	if (HasEntityEffect(cEntityEffect::effMiningFatigue))
+	{
+		int intensity = GetEntityEffect(cEntityEffect::effMiningFatigue)->GetIntensity();
+		switch (intensity)
+		{
+		case 0:
+			f *= 0.3f;
+			break;
+		case 1:
+			f *= 0.09f;
+			break;
+		case 2:
+			f *= 0.0027f;
+			break;
+		default:
+			f *= 8.1e-4f;
+		}
+	}
+
+	if ((IsInsideOfBlock(E_BLOCK_WATER) || IsInsideOfBlock(E_BLOCK_STATIONARY_WATER)) && !(GetEquippedItem().m_Enchantments.GetLevel(cEnchantments::eEnchantment::enchAquaAffinity) > 0))
```

Comment:
Wrap this `if` statement onto multiple lines based on the "tree" of operations.
How about lava, doesn't that impair digging speed as well?